### PR TITLE
Testing helper support `Note`

### PR DESCRIPTION
**Pull request by shaopeng-gh**
This change only adds the support `Note` for test helpers.

### CODE REVIEW

File: src/Test.FunctionalTests.BinSkim.Rules/TestMessageLogger.cs [hunk 1]
```diff
@@ -25,6 +26,8 @@ public TestMessageLogger()
 
         public HashSet<string> FailTargets { get; set; }
 
+        public HashSet<string> FailWithNoteTargets { get; set; }
```
Comment (michaelcfanning) [inline src/Test.FunctionalTests.BinSkim.Rules/TestMessageLogger.cs:29]: >FailWithNoteTargets [](http://example.com/codeflow?start=31&length=19)

This should be named 'Note' targets

**Comment by michaelcfanning** [comment]
>                     this.FailTargets.Add(targetPath);

We should differentiate warnings from errors.

ErrorTargets
WarningTargets
NoteTargets

Otherwise this is all confusing.

---
Refers to: src/Test.FunctionalTests.BinSkim.Rules/TestMessageLogger.cs:89 in ede038e. [](commit_id = ede038e2ecc5ca6eca3bbcae6c43827712d2a225, deletion_comment = False)

**Comment by shaopeng-gh** [comment]
>                     this.FailTargets.Add(targetPath);

sounds good, working on the fix.

---
In reply to: [1440801772](https://github.com/microsoft/binskim/pull/809#issuecomment-1440801772) [](http://example.com/codeflow?ancestors=1440801772,1440801772)

---
Refers to: src/Test.FunctionalTests.BinSkim.Rules/TestMessageLogger.cs:89 in ede038e. [](commit_id = ede038e2ecc5ca6eca3bbcae6c43827712d2a225, deletion_comment = False)

**Comment by shaopeng-gh** [comment]
>                     this.FailTargets.Add(targetPath);

fixed

---
In reply to: [1441102486](https://github.com/microsoft/binskim/pull/809#issuecomment-1441102486) [](http://example.com/codeflow?ancestors=1441102486,1440801772,1440801772)

---
Refers to: src/Test.FunctionalTests.BinSkim.Rules/TestMessageLogger.cs:89 in ede038e. [](commit_id = ede038e2ecc5ca6eca3bbcae6c43827712d2a225, deletion_comment = False)

File: src/Test.FunctionalTests.BinSkim.Rules/TestMessageLogger.cs (hunk 1 above)
Comment (shaopeng-gh) [reply src/Test.FunctionalTests.BinSkim.Rules/TestMessageLogger.cs]: fixed

**Comment by shaopeng-gh** [comment]
>         public void NoteTestResult(FailureLevel messageKind, string targetPath)

fixed

---
In reply to: [1440802015](https://github.com/microsoft/binskim/pull/809#issuecomment-1440802015) [](http://example.com/codeflow?ancestors=1440802015,1440802015)

---
Refers to: src/Test.FunctionalTests.BinSkim.Rules/TestMessageLogger.cs:77 in ede038e. [](commit_id = ede038e2ecc5ca6eca3bbcae6c43827712d2a225, deletion_comment = False)

**Review by michaelcfanning — APPROVED** [review]
:shipit:

**Pull request merged**